modules/openapi-generator/src/main/resources/go-experimental/configuration.mustache

```diff
@@ -144,3 +203,84 @@ func (c *Configuration) ServerUrl(index int, variables map[string]string) (strin
 	}
 	return url, nil
 }
+
+// ServerURL returns URL based on server settings
+func (c *Configuration) ServerURL(index int, variables map[string]string) (string, error) {
+	return c.Servers.URL(index, variables)
+}
+
+func getServerIndex(ctx context.Context) (int, error) {
+	si := ctx.Value(ContextServerIndex)
+	if si != nil {
+		if index, ok := si.(int); ok {
+			return index, nil
+		}
+		return 0, reportError("Invalid type %T should be int", si)
```

Comment:
It would be nice to use something like "ctx value of ContextServerIndex has invalid type ..." so that it would be immediately obvious from the error where the problem is.